src/opentimelineview/track_widgets.py

```diff
@@ -239,15 +241,54 @@ class TrackNameItem(BaseItem):
     def __init__(self, track, rect, *args, **kwargs):
         super(TrackNameItem, self).__init__(None, None, rect,
                                             *args, **kwargs)
-        track_name = 'Track' if track.name == '' else track.name
-        if len(track_name) > 7:
-            track_name = track_name[:7] + '...'
-        self.source_name_label.setText(track_name + '\n({})'.format(track.kind))
+        self.track = track
+        self.track_name = 'Track' if track.name == '' else track.name
```

Comment:
```suggestion
        self.track_name = 'Track' if not track.name else track.name
```
Should be the same but also covers the case where `track.name` is `None`